_posts/en/newsletters/2024-01-24-newsletter.md

```diff
@@ -0,0 +1,240 @@
+    of any code that might be used to validate BIP68 relative locktimes
+    are urged to check the code for the same flaw.
+
+- **Proposed changes to LN for v3 relay and ephemeral anchors:** Bastien
+  Teinturier [posted][teinturier v3] to Delving Bitcoin to describe the
+  changes he thinks should be made to the LN specification to make
+  effective use of [v3 transaction relay][topic v3 transaction relay]
+  and [ephemeral anchors][topic ephemeral anchors].  The changes appear
+  simple:
+
+  - *Anchor swap:* the commitment transaction's two current [anchor outputs][topic
+    anchor outputs] used to guarantee the ability to [CPFP fee bump][topic cpfp]
+    under the [CPFP carve out][topic cpfp carve out] policy are removed
+    and one ephemeral anchor is added in their place.
+
+  - *Reducing delays:* the extra 1-block delays on the commitment transaction are removed.
+    They were added to ensure that CPFP carve out would always work but are
+    not necessary under the v3 relay policy.
+
+  - *Trimming redirect:* instead of spending the value of all [trimmed HTLCs][topic trimmed
+    htlc] to fees in the commitment transaction, the combined value is
+    added to the value of the anchor output, ensuring the extra fees are
+    used to guarantee confirmation of both the commitment and the ephemeral anchor
+    (see [last week's newsletter][news285 mev] for discussion).
+
+  - *Other changes:* some other minor changes and simplifications are made.
+
+  The subsequent discussion examined several interesting consequences of
+  the proposed changes, including:
+
+  - *Reduced UTXO requirements:* ensuring the correct channel state goes onchain becomes easier
+    thanks to the removal of the extra 1-block delay.  If a faulty party
+    broadcasts a revoked commitment transaction, the other party can use
+    their main output from that commitment to CPFP fee bump that revoked
+    commitment.  They don't need to keep a separate confirmed UTXO for
+    that purpose.  For this to be safe, the party must keep a sufficient
+    reserve balance, as the 1% minimum specified in [BOLT2][] may not be
+    enough to cover fees.  For a non-forwarding node that keeps a
+    sufficient reserve, the only time they need a separate UTXO for
+    security reasons is when they want to accept an incoming payment.
+
+  - *Circular dependencies:* v3 relay depends on [cluster mempool][topic
+    cluster mempool], cluster mempool depends on CPFP carve out being
+    removed, but carve out can't be removed until LN users switch to v3
+    relay.  Gregory Sanders [proposes][sanders transition] an
+    intermediate stage with temporary special treatment of transactions
+    that look like current anchors-style LN commitment transactions,
+    allowing Bitcoin Core to deploy cluster mempool and v3.  When
+    they've been widely deployed and all LN implementations have
+    upgraded to using them, the temporary special rules can be dropped.
```

Comment:
(Apologies for the late review) proposed new wording:

```suggestion
  - *Imbued v3 logic:* In response to concerns voiced in the LN spec meeting that it may
    take a long time for LN to design, implement, and deploy these changes,
    Gregory Sanders [proposed][sanders transition] an
    intermediate stage with temporary special treatment of transactions
    that look like current anchors-style LN commitment transactions,
    allowing Bitcoin Core to deploy cluster mempool without being blocked by LN development.  When
    they've been widely deployed and all LN implementations have
    upgraded to using v3, the temporary special rules can be dropped.
```

Reply:
@glozow thanks for catching this error and providing the rewording!

(For anyone who's curious, I made a bad inference from this sentence by Greg: "We need to roll out **updates to mempool/relay** to make the transition to v3 style transactions safe, so LN implementations can adopt it" (emphasis added).  I inferred the "updates to mempool" meant cluster mempool.  I'm guessing now that what he really meant was more in the line of package relay / package rbf.)

Reply:
Understandable. It's my bad I should have reviewed earlier. Thanks @bitschmidty for being flexible and making last-minute changes!